### ISSUE TITLE
bug: `props-must-be-readonly` reports error although mutable is set when attribute is set via constant value

### ISSUE THREAD

**Issue by pinussilvestrus** [ionic-team/stencil-eslint#196]
### Prerequisites

- [x] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [x] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/stenciljs/eslint-plugin/issues) that already report this problem, without success.

### Stencil ESLint Version

1.2.0

### Current Behavior

When declaring a `@Prop` via enum or constant, the `props-must-be-readonly` is reporting an error although `mutable` was set to `true`.

```tsx
@Prop({ reflect: true, attribute: Attribute.withPadding, mutable: true }) withPadding: boolean = false;
```

=> 

```sh
  26:77  error  Class properties decorated with @Prop() should be readonly  stencil/props-must-be-readonly
```

In comparison, when setting the `attribute` directly without an enum or const, it works as expected (no error is reported)

```tsx
 @Prop({ reflect: true, attribute: 'with-padding', mutable: true }) withPadding: boolean = false;
```

### Expected Behavior

The `@Prop` declaration

```tsx
@Prop({ reflect: true, attribute: Attribute.withPadding, mutable: true }) withPadding: boolean = false;
```

should not throw an error, as `mutable` is set to `true` (cf. [docs](https://github.com/stenciljs/eslint-plugin/blob/main/docs/props-must-be-readonly.md#props-must-be-readonly)).

### Steps to Reproduce

* Declare a property as above
* Run the linting rules with `props-must-be-readonly` enabled

### Code Reproduction URL

https://github.com/pinussilvestrus/test-stencil/commit/0d943e601dd76a51bd819527802f6d6d4f77985c
